### PR TITLE
Render special chars in variables' default value properly

### PR DESCRIPTION
**Pull request by julienduchesne**
### Prerequisites

Put an `x` into the box(es) that apply:

- [x] This pull request fixes a bug.
- [ ] This pull request adds a feature.
- [ ] This pull request enhances existing functionality.
- [ ] This pull request introduces breaking change.

For more information, see the [Contributing Guide](https://github.com/segmentio/terraform-docs/tree/master/CONTRIBUTING.md).

### Description

Check out the tests. A variable with special chars would just give 
`Error: json: error calling MarshalJSON for type types.String: invalid character '.' in string escape code` and panic

### Checklist

Put an `x` into all boxes that apply:

- [x] I have read the [Contributing Guidelines](https://github.com/segmentio/terraform-docs/tree/master/CONTRIBUTING.md).

#### Tests

- [x] I have added tests to cover my changes.
- [x] All tests pass when I run `make test`.

#### Documentation

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

#### Code Style

- [x] My code follows the code style of this project.

### CODE REVIEW

**Comment by khos2ow** [comment]
fixes #238 
related to #259

**Comment by julienduchesne** [comment]
Thanks for the review!

Here you go. I changed the definition of `GetValue` to not escape the chars: https://github.com/terraform-docs/terraform-docs/pull/284/files#diff-0dab69e5c2f94db8fa8283c242661b63R24. For JSON, it doesn't go through a template so whether or not it's escaped depends on the EscapeCharacters settings. For TFVars JSON, it is always escaped it seems

As for your second question, I do not know how I would fix that. My immediate issue is that terraform-docs just doesn't work if there's a special char in a variable. If a user wants a regex in a variable, he can unescape it IMO

File: docs/formats/tfvars-json.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ generates the following output:
       "string-1": "bar",
       "string-2": null,
       "string-3": "",
+      "string-special-chars": "\\.\u003c\u003e[]{}_-",
```
Comment (khos2ow) [inline docs/formats/tfvars-json.md:96]: This is a very valid point and after a little bit of digging I found that `orderedMap` doesn't allow for escaping which I sent them a PR (https://github.com/iancoleman/orderedmap/pull/15). We can proceed with this PR here and adjust accordingly when the other PR gets processed.

File: go.mod [hunk 2]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/hashicorp/hcl/v2 v2.6.0
 	github.com/iancoleman/orderedmap v0.0.0-20190318233801-ac98e3ecb4b0
 	github.com/imdario/mergo v0.3.9
+	github.com/segmentio/terraform-docs v0.9.1 // indirect
```
[the remark this answers is not in the document]
Comment (julienduchesne) [reply go.mod]: Done!

**Pull request merged**